### ISSUE TITLE
Hackathon Embedded Registration Flow

### ISSUE THREAD

**Issue by frankchen07** [gitcoinco/web#6087]
### User Story

As a Gitcoin admin, I want an embedded survey for each Gitcoin hackathon so I know the motivations of each hacker

### Why Is this Needed

In previous hackathons, we employed post and even pre hack surveys asking several open-ended questions. Response rates would usually be <50% and the surveys were too open-ended, prolonging the user experience.

### Description

During hackathon registration, we add an embedded, multiple choice survey that is simple for the user to reveal their intentions for the hackathon. This eliminates the need for typeform & google document surveys before and after the hackathon, and still provides a high-level level of information on Gitcoin hackers to us.

### Current Behavior

Currently, there is not an embedded survey. Any surveys pre or post are sent out separately from the Gitcoin app.

### Expected Behavior

Add an embedded survey simply asking what a hacker's motivation for the hackathon is.

### Designs & Flows

User flows on Figma: https://www.figma.com/file/2qWGMbfd3z1txPndLNAZ9m/Untitled?node-id=0%3A1

**Details**

<img width="1095" alt="Screen Shot 2020-02-26 at 15 07 05" src="https://user-images.githubusercontent.com/7516920/75396632-d6855100-58a9-11ea-87bd-105386ab8e35.png">

1. The title should be dynamic and be correct for the hackathon that is currently or will be running.
2. The hackathon start and end dates should be dynamic and correct as well.

<img width="808" alt="Screen Shot 2020-02-26 at 15 03 54" src="https://user-images.githubusercontent.com/7516920/75396484-6ecf0600-58a9-11ea-9679-385b97ff73cf.png">

1. The end timer should be counting down to the hackathon start date.
2. The 4-items checklist should be check marks all the way down, and those should be links that open a new tab to the correct page.
    a. townsquare = gitcoin.co
    b. workshops = TBD
    c. sponsors below = should lead to the organization page
    d. invite fiends = TBD

<img width="1074" alt="Screen Shot 2020-02-26 at 15 04 13" src="https://user-images.githubusercontent.com/7516920/75396644-dc7b3200-58a9-11ea-8f22-89141c159446.png">

1. Sponsors that show up should be the hackathon sponsors that are or will be in the hackathon that the user is signing up for
2. Sponsor link, when clicked, should lead to the organization page (e.g gitcoin.co/gitcoinco)

### Definition of Done
- [ ] new survey section built into existing hackathon signup flow
- [ ] details of the new hackathon page (on figma) built out to specifications

### Data Requirements
- [ ] ensure that the "reason for joining the hackathon" response is recorded in the data model

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__This issue now has a funding of 1.5 ETH (331.64 USD @ $221.09/ETH)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/gitcoinco/web/6087/4073).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/gitcoinco/web/6087/4073).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $84,973.41 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 266 years, 8 months from now.
Please review their action plans below:


**1) [zoek1](https://gitcoin.co/zoek1) has started work.**

I'll implement a survey mechanism to collect feedback from hackathons users.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/gitcoinco/web/6087/4073).

**Comment by frankchen07** [comment]
@zoek1 - awesome, when can we expect a WIP PR? :)

**Comment by zoek1** [comment]
Between today or tomorrow @frankchen07 :wink:

**Comment by zoek1** [comment]
Hey @frankchen07, i pushed changes to build the survey from the admin panel. My next step is to implement the UI changes.

**Comment by zoek1** [comment]
Still working on this more updates in a couple of hours :smile:

**Comment by frankchen07** [comment]
Awesome - thanks for the updates! When you're almost done, if it's not too hard, can you make a screen video of the changes so we can also see them?

**Comment by frankchen07** [comment]
@zoek1 - hi Miguel - where are we on this embedded onboarding ticket?

**Comment by zoek1** [comment]
Only i need to work on the UI, the lógic is already implemented. Sorry I've been postponing this task, but this is my priority this week :pray:

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work for 1.5 ETH (254.86 USD @ $169.91/ETH) has been submitted by__:

 1. [@zoek1](https://gitcoin.co/zoek1)



@frankchen07 please take a look at the submitted work:
* [PR](https://github.com/gitcoinco/web/pull/6128) by @zoek1

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/gitcoinco/web/6087/4073)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/gitcoinco/web/6087/4073).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $87,379.07 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
<table>
<tr>
<td>
<a title='Django Cowboy' href='https://gitcoin.co/kudos/146/django_cowboy'> <img width='250' src='https://gitcoin.co/dynamic/kudos/146/django_cowboy' alt='Django Cowboy'> </a> 
</td>
<td>
⚡️ A *Django Cowboy* Kudos has been sent to @zoek1 for this issue from @frankchen07. ⚡️ 

The sender had the following public comments: 
> hey Miguel - great work on this task, jumping on it and bringing it all the way to completion!

Nice work @zoek1! 
Your Kudos has automatically been sent in the ETH address we have on file.
</td>
</tr>
</table>

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__The funding of 1.5 ETH (321.26 USD @ $214.17/ETH)  attached to this issue has been approved & issued to @zoek1.__  

 
* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/gitcoinco/web/6087/4073)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $86,311.50 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by alexvotofuture** [comment]
Awesome stuff! Have some fix requests now that I've seen it in production:
-Town Square link should go to that hackathon’s Town Square tab
-Remove Town Square kudos promise in parenthesis - we haven’t created a process around this yet
-There’s no way to RSVP for workshops yet, so this line can just say “Look out for upcoming workshop invites in your email inbox”
-Sponsors link is dead. Remove this for now?
-Invite your friends to hack - @vivek did we have a referral program somewhere? If not, should we remove this?

Would love to see this page rolled into a tab on the hackathon explorer (without re-directing the user to a new page, as is currently the case with the onboard page).

**Comment by octavioamu** [comment]
@PixelantDesign @frankchen07 just an important note, this ticket and design file wasn't contemplating the actual data on that view, so all the area of hackathon description and the ability to edit was removed. 
Im adding that back since is where we can explain special conditions for each hackathon. 
Also Im adding the logos from the associated hackathons sponsors fields instead of sponsors model as was proposed on the hackathon data changes we did.
